README.md

```diff
@@ -193,6 +194,24 @@ fastify.register(require('fastify-rate-limit'), {
 })
 ```
 
+Custom `hook` example usage (after authentication):
+```js
+fastify.register(require('fastify-rate-limit'), {
+  hook: 'preHandler',
+  keyGenerator: function(req) {
+    return return req.userId || req.ip
+  }
+})
+
+fastify.decorateRequest('userId', '')
+fastify.addHook('preHandler', async req => {
+  const { userId } = req.query
+  if (userId) {
+    req.userId = userId
```

Comment:
```suggestion
  keyGenerator: function (request) {
    return request.userId || request.ip
  }
})

fastify.decorateRequest('userId', '')
fastify.addHook('preHandler', async function (request) {
  const { userId } = request.query
  if (userId) {
    request.userId = userId
```

As a "convention" we use `request` over `req` and `reply` over `res` across all fastify org repositories.

Reply:
Personally I also use `request`, but in the examples here `req` was more frequent, so I followed...